internal/locate/region_cache_test.go

```diff
@@ -1918,6 +1918,124 @@ func (s *testRegionCacheSuite) TestLocateBucket() {
 	}
 }
 
+// TestBucketClampingToRegion tests that bucket boundaries exceeding region
+// boundaries are clamped correctly.
+func TestBucketClampingToRegion(t *testing.T) {
+	tests := []struct {
+		name            string
+		regionStart     []byte
+		regionEnd       []byte
+		bucketStart     []byte
+		bucketEnd       []byte
+		wantStart       []byte
+		wantEnd         []byte
+		shouldBeClamped bool
+	}{
+		{
+			name:            "bucket within region - no clamping",
+			regionStart:     []byte("a"),
+			regionEnd:       []byte("z"),
+			bucketStart:     []byte("f"),
+			bucketEnd:       []byte("m"),
+			wantStart:       []byte("f"),
+			wantEnd:         []byte("m"),
+			shouldBeClamped: false,
+		},
+		{
+			name:            "bucket start before region - clamp start",
+			regionStart:     []byte("f"),
+			regionEnd:       []byte("z"),
+			bucketStart:     []byte("a"), // Before region start
+			bucketEnd:       []byte("m"),
+			wantStart:       []byte("f"), // Clamped to region start
+			wantEnd:         []byte("m"),
+			shouldBeClamped: true,
+		},
+		{
+			name:            "bucket end after region - clamp end",
+			regionStart:     []byte("a"),
+			regionEnd:       []byte("m"),
+			bucketStart:     []byte("f"),
+			bucketEnd:       []byte("z"), // After region end
+			wantStart:       []byte("f"),
+			wantEnd:         []byte("m"), // Clamped to region end
+			shouldBeClamped: true,
+		},
+		{
+			name:            "bucket exceeds both boundaries - clamp both",
+			regionStart:     []byte("f"),
+			regionEnd:       []byte("m"),
+			bucketStart:     []byte("a"), // Before region start
+			bucketEnd:       []byte("z"), // After region end
+			wantStart:       []byte("f"), // Clamped to region start
+			wantEnd:         []byte("m"), // Clamped to region end
+			shouldBeClamped: true,
+		},
+		{
+			name:            "empty region start - bucket start not clamped",
+			regionStart:     []byte{}, // Beginning of keyspace
+			regionEnd:       []byte("m"),
+			bucketStart:     []byte("a"),
+			bucketEnd:       []byte("z"),
+			wantStart:       []byte("a"),
+			wantEnd:         []byte("m"), // Only end clamped
+			shouldBeClamped: true,
+		},
+		{
+			name:            "empty region end - bucket end not clamped",
+			regionStart:     []byte("f"),
+			regionEnd:       []byte{}, // End of keyspace (infinity)
+			bucketStart:     []byte("a"),
+			bucketEnd:       []byte("z"),
+			wantStart:       []byte("f"), // Only start clamped
+			wantEnd:         []byte("z"),
+			shouldBeClamped: true,
+		},
+		{
+			name:            "bucket completely before region - fallback to region",
+			regionStart:     []byte("m"),
+			regionEnd:       []byte("z"),
+			bucketStart:     []byte("a"),
+			bucketEnd:       []byte("f"), // Ends before region starts
+			wantStart:       []byte("m"), // Falls back to region boundaries
+			wantEnd:         []byte("z"),
+			shouldBeClamped: true,
+		},
```

Comment:
Looks there better be some cases where the bucket's keys has empty values, combined with the region's range keys either containing or not containing empty values.